packages/common/src/index-api.ts

```diff
@@ -24,13 +24,76 @@ export type BackwardPagination = {
  */
 export type Pagination = ForwardPagination | BackwardPagination
 
+/**
+ * Boolean field value filter
+ */
+export type BooleanValueFilter = { isNull: boolean } | { equalTo: boolean }
+
+/**
+ * Enum field value filter
+ */
+export type EnumValueFilter =
+  | { isNull: boolean }
+  | { equalTo: string }
+  | { notEqualTo: string }
+  | { in: Array<string> }
+  | { notIn: Array<string> }
+
+/**
+ * String or number field value filter
+ */
+export type ScalarValueFilter<T extends string | number = string | number> =
```

Comment:
sorry my typescript knowledge isn't good enough to understand what this is saying, can you break it down for me?

Reply:
- `T extends string | number` restricts the valid types for `T`, so `ScalarValueFilter<string>` is valid, but `ScalarValueFilter<boolean>` is invalid.
- `T = string | number` defines a default value for `T` when not defined, so `ScalarValueFilter` can be used without explicitly providing the type for `T`, using `const filter: ScalarValueFilter` is the same as using `const filter: ScalarValueFilter<string | number>`.